lib/Basic/ExecutionQueue.cpp

```diff
@@ -39,14 +39,11 @@ ExecutionQueue::~ExecutionQueue() {
 
 ProcessStatus ExecutionQueue::executeProcess(QueueJobContext* context,
                                              ArrayRef<StringRef> commandLine) {
-  // Promises are move constructible only, thus cannot be put into std::function
-  // objects that themselves get copied around. So we must create a shared_ptr
-  // here to allow it to go along with the labmda.
-  std::shared_ptr<std::promise<ProcessStatus>> p{new std::promise<ProcessStatus>};
-  auto result = p->get_future();
-  executeProcess(context, commandLine, {}, true, {true},
-                 {[p](ProcessResult result) mutable {
-    p->set_value(result.status);
+  std::promise<ProcessStatus> p;
```

Comment:
Hah, duh.  The promise doesn't actually escape.

Reply:
If the promise doesn’t actually escape, then doesn’t this code actually not even need a promise/future?